### PR TITLE
add BayesPrism to omnideconv

### PR DESCRIPTION
**Pull request by alex-d13**
Repository: https://github.com/Danko-Lab/TED
Publication: https://doi.org/10.1038/s43018-022-00356-3

Currently needs more testing with different datasets and the imports have to be checked again.

### CODE REVIEW

**Comment by alex-d13** [comment]
BayesPrism is tested and could be merged.
I tested it with one external dataset and the sample dataset.

@grst maybe review the changes?
I am not sure if this should belong into the function itself or should be placed somewhere else https://github.com/omnideconv/omnideconv/pull/112/commits/4786967acbb837865983fa7fe2d7c19bc7f04aae#diff-e7af15b7e6f7dd0ffc4b37e52d9b70624a0ff3ba7d7d22942ad595b25f928636R89

File: R/BayesPrism.R [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+#'
+deconvolute_bayesprism <- function(bulk_gene_expression, single_cell_object, cell_type_annotations,
+                                   cell_subtype_labels = NULL, tum_key = NULL, pseudo_min = 1E-8, alpha = 1,
+                                   sigma = 2, outlier_cut = 0.01, outlier_fraction = 0.1,
+                                   gibbs_control = list(chain.length = 1000, burn.in = 500, thinning = 2),
+                                   opt_control = list(trace = 0, maxit = 100000), n_cores = 1,
+                                   n_cores_2g = NULL, first_gibbs_only = FALSE, seed = NULL) {
+
+  # package is required to run bayesprism, also with only a single core
+  require(snowfall)
```
Comment (grst) [inline R/BayesPrism.R:89]: in a package one should never use `require` or `library`. 

instead use `@import` or `@importFrom` as a roxygen comment

**Review by grst — COMMENTED** [review]
I've seen that you have the `require(snowfall)`, but you're also not using this package directly, right? 

If the package really needs to be attached manually for `run.Ted` to work it's probably an issue with their code. In that case we can either try to fix it upstream or use `require` as you do.

**Comment by alex-d13** [comment]
The last commit includes an update, where I use the code from the correct repository (Danko-Lab/BayesPrism/BayesPrism), which also fixed the issue with the `snowfall` package from above.
Should we also fork this repository into our omnideconv organisation?

**Comment by grst** [comment]
I would only fork repos that we patched and couldn't get the authors to
merge the patch upstream.

**Comment by grst** [comment]
CI is fixed now on main.

**Pull request merged**